nbs/13_cog_sights.ipynb

```diff
@@ -0,0 +1,426 @@
+      "  self.chain = LLMChain(llm=self.llm, prompt=self.prompt)\n"
+     ]
+    }
+   ],
+   "source": [
+    "import json\n",
+    "import re\n",
+    "from datetime import datetime\n",
+    "from typing import Any, Dict, List, Optional, Tuple, TypedDict, Union\n",
+    "\n",
+    "import boto3\n",
+    "import numpy as np\n",
+    "import rasterio\n",
+    "import xarray as xr\n",
+    "from pystac_client import Client\n",
+    "from rasterio.mask import mask\n",
+    "from rasterio.session import AWSSession\n",
+    "from shapely.geometry import shape\n",
+    "from stackstac import stack\n",
+    "from pydantic import BaseModel\n",
+    "\n",
+    "from langchain.chains import LLMChain\n",
+    "from langchain_openai import ChatOpenAI\n",
+    "from langchain.prompts import ChatPromptTemplate, PromptTemplate\n",
+    "from langchain.schema import BaseOutputParser\n",
+    "from langchain.tools import tool\n",
+    "from langchain_core.runnables import RunnableLambda\n",
+    "from langgraph.graph import StateGraph, END\n",
+    "\n",
+    "\n",
+    "NL_classification_values = {\n",
+    "    2: \"natural forests\",\n",
+    "    3: \"natural short vegetation\",\n",
+    "    4: \"natural water\",\n",
+    "    5: \"mangroves\",\n",
+    "    6: \"bare\",\n",
+    "    7: \"snow\",\n",
+    "    8: \"wet natural forests\",\n",
+    "    9: \"natural peat forests\",\n",
+    "    10: \"wet natural short vegetation\",\n",
+    "    11: \"natural peat short vegetation\",\n",
+    "    12: \"crop\",\n",
+    "    13: \"built\",\n",
+    "    14: \"non-natural tree cover\",\n",
+    "    15: \"non-natural short vegetation\",\n",
+    "    16: \"non-natural water\",\n",
+    "    17: \"wet non-natural tree cover\",\n",
+    "    18: \"non-natural peat tree cover\",\n",
+    "    19: \"wet non-natural short vegetation\",\n",
+    "    20: \"non-natural peat short vegetation\",\n",
+    "    21: \"non-natural bare\"\n",
+    "}\n",
+    "\n",
+    "DEFAULT_COG_PATH = \"s3://gfw-data-lake/umd_glad_dist_alerts/v20250329/raster/epsg-4326/cog/default.tif\"\n",
+    "INTENSITY_COG_PATH = \"s3://gfw-data-lake/umd_glad_dist_alerts/v20250329/raster/epsg-4326/cog/intensity.tif\"\n",
+    "STAC_API_URL = \"https://eoapi.zeno-staging.ds.io/stac\"\n",
+    "AWS_PROFILE = \"zeno_internal_sso\"\n",
+    "REFERENCE_DATE = datetime(2015, 1, 1)\n",
+    "\n",
+    "def convert_date_range_to_days(date_range):\n",
+    "    start_date = datetime.strptime(str(date_range[0])[:10], \"%Y-%m-%d\")\n",
+    "    end_date = datetime.strptime(str(date_range[1])[:10], \"%Y-%m-%d\")\n",
+    "    return (start_date - REFERENCE_DATE).days, (end_date - REFERENCE_DATE).days, start_date, end_date\n",
+    "\n",
+    "def load_alert_data(geometry):\n",
```

Comment:
We should put the dist alerts into STAC as well and use a single tool for data retrieval, independent of the collection (see below).